lua/entities/gmod_wire_expression2/core/custom/cfc_e2_lib.lua

```diff
@@ -0,0 +1,138 @@
+E2Lib.RegisterExtension( "cfc_e2_lib", true )
+
+-- PvP Functions
+e2function number entity:playerIsInPvp()
+    if not IsValid(this) or not this:IsPlayer() then return 0 end
+    if this:GetNWBool( "CFC_PvP_Mode", false ) == true then return 1 end
+
+    return 0
+end
+
+e2function number entity:playerIsInBuild()
+    if not IsValid(this) or not this:IsPlayer() then return 0 end
+    if this:GetNWBool( "CFC_PvP_Mode", false ) == false then return 1 end
+
+    return 0
+end
+
+e2function string entity:getPvpStatus()
+    if not IsValid(this) or not this:IsPlayer() then return "" end
+    if this:GetNWBool( "CFC_PvP_Mode", false ) == true then return "pvp" end
+
+    return "build"
+end
+
+e2function table getPvpers()
+    local pvpers = {}
+    for _, v in pairs(player.GetHumans()) do
+        if v:GetNWBool( "CFC_PvP_Mode", false ) then pvpers[#pvpers + 1] = v end
+    end
+
+    return pvpers
+end
+
+e2function table getBuilders()
+    local builders = {}
+    for _, v in pairs(player.GetHumans()) do
+        if v:GetNWBool( "CFC_PvP_Mode", false ) == false then builders[#builders + 1] = v end
+    end
+
+    return builders
+end
+
+-- Ping Functions
+e2function number averagePing()
+    local sum = 0
+    local humans = player.GetHumans()
+    for _, v in pairs( humans ) do
+        sum = sum + v:Ping()
+    end
+	local avg = sum / table.Count( humans )
+
+    return avg
+end
+
+-- String Functions
+e2function number string:startsWith(string start)
+    if string.StartWith( this, start ) then return 1 end
+
```

Comment:
I probably commented here earlier, but if it's only two lines I think it's fine without a newline before the default return.